Content/WorldGeneration/GenerateLootWraiths.cs

```diff
@@ -0,0 +1,28 @@
+﻿using StarlightRiver.Content.NPCs.Misc;
+using Terraria.DataStructures;
+
+namespace StarlightRiver.Core
+{
+	public partial class StarlightWorld : ModSystem
+	{
+		private void LootWraithGen()
+		{
+			foreach (Chest chest in Main.chest)
+			{
+				if (chest != null && Main.rand.NextBool(15))
```

Comment:
Should we maybe yield the random check untill after we determine this is the correct type of chest? Seems redundant to roll it on every chest then check the type afterwards

Reply:
Also, we should be using the world gen random here, not Main.rand so that this behavior is consistent across seeds!